src/integer/mat_z/vector/dot_product.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module includes functionality to compute the dot product of two vectors.
+
+use crate::error::MathError;
+use crate::integer::{MatZ, Z};
+use crate::traits::{GetNumColumns, GetNumRows};
+use flint_sys::fmpz::fmpz_addmul;
+
+impl MatZ {
+    /// Returns the dot product of two vectors of type [`MatZ`].
+    ///
+    /// Paramters:
+    /// - `other`: specifies the other vector the dot product is calculated over
+    ///
```

Comment:
If we follow the style guide completely, we miss the return statement here

Reply:
👍 Correct, I added the note for returned values.